apps/admin-gui/src/app/shared/components/assign-groups-sponsored-members/assign-groups-sponsored-members-component.ts

```diff
@@ -0,0 +1,188 @@
+import { Component, EventEmitter, Input, OnDestroy, OnInit, Output } from '@angular/core';
+import { SelectionModel } from '@angular/cdk/collections';
+import {
+  AttributesManagerService,
+  Group,
+  GroupsManagerService,
+  RichGroup,
+} from '@perun-web-apps/perun/openapi';
+import { GuiAuthResolver } from '@perun-web-apps/perun/services';
+import { TABLE_VO_GROUPS } from '@perun-web-apps/config/table-config';
+import { Observable, Subscription } from 'rxjs';
+import { Urns } from '@perun-web-apps/perun/urns';
+import { hasBooleanAttributeEnabled, isGroupSynchronized } from '@perun-web-apps/perun/utils';
+
+@Component({
+  selector: 'app-assign-groups-sponsored-members-component',
+  templateUrl: 'assign-groups-sponsored-members-component.html',
+  styleUrls: ['assign-groups-sponsored-members-component.scss'],
+})
+export class AssignGroupsSponsoredMembersComponent implements OnInit, OnDestroy {
+  @Input() submit: Observable<void>;
+
+  @Input() voId: number;
+  @Output() groupsToAdd = new EventEmitter<number[]>();
+  @Output() submitAllowed = new EventEmitter<boolean>();
+
+  loading = false;
+  createGroupAuth: boolean;
+  assignableGroups: Group[] = [];
+  allVoGroups: Group[] = [];
+  manualMemberAddingBlocked = false;
+  groupAssignment = 'none';
+
+  selection = new SelectionModel<Group>(true, []);
+  name = '';
+  description = '';
+  asSubgroup = false;
+  parentGroup: Group = null;
+  filterValue = '';
+  tableId = TABLE_VO_GROUPS;
+  groupToCreate: Group = {
+    id: 0,
+    beanName: 'Group',
+  };
+  groupIds: number[] = [];
+  onSubmit: Subscription;
+
+  private groupAttrNames = [Urns.GROUP_SYNC_ENABLED, Urns.GROUP_BLOCK_MANUAL_MEMBER_ADDING];
+
+  constructor(
+    private groupsService: GroupsManagerService,
+    private guiAuthResolver: GuiAuthResolver,
+    private attributesService: AttributesManagerService
+  ) {}
+
+  ngOnInit(): void {
+    this.submit.subscribe(() => this.handleGroups());
```

Comment:
You probably forgot to assign the subscription to the property you are trying to unsubscribe in the `ngOnDestroy` block. I tested it and after this assignment, it looks better.
```suggestion
    this.onSubmit = this.submit.subscribe(() => this.handleGroups());
```